### ISSUE TITLE
Keyboard navigation shortcuts

### ISSUE THREAD

**Issue by ghost** [servo/servo#45904]
**Describe the new feature:**
According to research mouse usage causes health issues [short term](https://www.tandfonline.com/doi/abs/10.1080/001401399184992) and [long term](https://pubmed.ncbi.nlm.nih.gov/21063306/). Ergonomic mouses help, but don't prevent the repetitive strain inherent to mice (https://pmc.ncbi.nlm.nih.gov/articles/PMC9265546/). Keyboard navigation is better and should be encouraged. Examples of such implementations: https://chromewebstore.google.com/detail/vimium-c-all-by-keyboard/hfjbmagddngcpeloejdejnfgbamkjaeg, https://qutebrowser.org/ and https://zed.dev/docs/vim. 

It would be nice if servo offers Vimium-C like keyboard navigation to improve not only the health, but also productivity for people. Since such a design would require less moving of the hands to do actions like clicking a link. And it would be consistent to what for example Chromium is offering of carret browsing. https://www.geeksforgeeks.org/techtips/what-is-caret-browsing-and-how-does-it-work/.

EDIT: One of the issues of just using an extension like Vimium-C is trust. Since you give it full permisions to your browser data, so a built-in solution like qutebrowser has done would be more secure.
